pkg/csi/service/common/vsphereutil.go

```diff
@@ -195,6 +195,26 @@ func CreateBlockVolumeUtil(ctx context.Context, clusterFlavor cnstypes.CnsCluste
 				Id: cnsSnapshotID,
 			},
 		}
+
+		// select the compatible datastore for the case of create volume from snapshot
+		// step 1: query the datastore of snapshot. By design, snapshot is always located at the same datastore
+		// as the source volume
+		cnsVolume, err := QueryVolumeByID(ctx, manager.VolumeManager, cnsVolumeID)
+		if err != nil {
+			return nil, logger.LogNewErrorf(log,
+				"failed to query datastore for the snapshot %s with error %+v",
```

Comment:
Shouldn't this be failed to query volume?

Reply:
`logger.LogNewErrorf` would return an error. My thinking here is, if `QueryVolumeByID` failed, the execution should not proceed anymore as we have no idea in which datastore to create the volume from snapshot.

Any concern here?

Reply:
I meant QueryVolumeByID failed, but the log message states "failed to query datastore for the snapshot", it may be confusing to users to see this error, perhaps state explicitly that query volume failed while determining compatible datastore?

Reply:
The QueryVolumeByID here is used to query the datastore of snapshot, since there is no way to directly query snapshot datastore and snapshot is always located in the same datastore as its source volume. So, the existing log message is expected.

BTW, the comment above the line clarifies it as well.
```
// step 1: query the datastore of snapshot. By design, snapshot is always located at the same datastore
		// as the source volume
```